api/src/server/event.rs

```diff
@@ -10,76 +10,138 @@ use tracing::debug;
         .first()
-        .ok_or_else(|| anyhow!("car data should have at least one root"))?;
+        .ok_or_else(|| anyhow!("CAR data should have at least one root"))?;
 
     debug!(%event_cid, "first root cid");
 
     let mut car_blocks = HashMap::new();
     while let Some((cid, bytes)) = car.next_block().await? {
         car_blocks.insert(cid, bytes);
     }
-    let event_bytes = get_block(&event_cid, &car_blocks, store).await?;
-    let event: unvalidated::RawEvent<Ipld> =
-        serde_ipld_dagcbor::from_slice(&event_bytes).context("decoding event")?;
+    let event_bytes = car_blocks
+        .get(&event_cid)
+        .ok_or_else(|| anyhow!("Event CAR data missing block for root CID"))?;
+    let raw_event: unvalidated::RawEvent<Ipld> =
+        serde_ipld_dagcbor::from_slice(event_bytes).context("decoding event")?;
+    // TODO(stbrody) add check for round-trip-ability to catch extra fields.
 
-    let (init_id, init_payload) = match event {
-        unvalidated::RawEvent::Time(event) => (
-            event.id(),
-            get_init_event_payload(&event.id(), &car_blocks, store).await?,
-        ),
-        unvalidated::RawEvent::Signed(event) => {
-            let link = event
+    match raw_event {
+        unvalidated::RawEvent::Time(event) => Ok((event_cid, unvalidated::Event::Time(event))),
+        unvalidated::RawEvent::Signed(envelope) => {
+            let payload_cid = envelope
                 .link()
                 .ok_or_else(|| anyhow!("event should have a link"))?;
 
-            let payload_bytes = get_block(&link, &car_blocks, store).await?;
+            let payload_bytes = car_blocks
+                .get(&payload_cid)
+                .ok_or_else(|| anyhow!("Signed Event CAR data missing block for payload"))?;
             let payload: unvalidated::Payload<Ipld> =
-                serde_ipld_dagcbor::from_slice(&payload_bytes).context("decoding payload")?;
-            let init_id = match payload {
-                unvalidated::Payload::Init(_) => event_cid,
-                unvalidated::Payload::Data(payload) => *payload.id(),
-            };
-            (
-                init_id,
-                get_init_event_payload(&init_id, &car_blocks, store).await?,
-            )
+                serde_ipld_dagcbor::from_slice(payload_bytes).context("decoding payload")?;
+            // TODO(stbrody) add check for round-trip-ability to catch extra fields.
+
+            Ok((
+                event_cid,
+                unvalidated::Event::Signed(unvalidated::signed::Event::new(
+                    event_cid,
+                    envelope,
+                    payload_cid,
+                    payload,
+                )),
+            ))
         }
-        unvalidated::RawEvent::Unsigned(event) => (event_cid, event),
-    };
+        unvalidated::RawEvent::Unsigned(event) => {
+            Ok((event_cid, unvalidated::Event::Unsigned(event)))
+        }
+    }
+}
 
+async fn event_id_for_event<S>(
```

Comment:
It might be nice to consolidate some of the behavior here with the service crate, which is now implementing `AccessModelStore` (aside: that should really be renamed to events or something) to have a shared `store -> &[u8] -> event` flow but no need to do it now.

Reply:
on the naming piece, I through together https://github.com/ceramicnetwork/rust-ceramic/pull/365 to rename the traits to InterestStore and EventStore.